src/bundle_builder.rs

```diff
@@ -0,0 +1,133 @@
+use rand::{distributions::Standard, Rng};
+use std::sync::Arc;
+
+use ethers::prelude::{signer::SignerMiddleware, *};
+use ethers_flashbots::BundleRequest;
+
+/// 1 kilobyte = 1024 bytes
+const KB: usize = 1024;
+
+/// Arbitrarily chosen number to cover for nonce+from+to+gas price size in a serialized
+/// transaction.  TODO: get the actual overhead from the signing, etc. to pack more fully
+const TRIM_BYTES: usize = 500;
+
+#[tracing::instrument(skip_all, name = "construct_bundle")]
+fn construct_tx(
+    chain_id: u64,
+    address: Address,
+    receiver: Address,
+    data_size: usize,
+    gas_price: U256,
+) -> ethers::prelude::TransactionRequest {
+    // Craft the transaction.
+    let blob = generate_random_data(data_size);
+
+    let tx = TransactionRequest::new()
+        .chain_id(chain_id)
+        .value(0)
+        .from(address)
+        .to(receiver)
+        .data(blob)
+        .gas_price(gas_price);
+
+    return tx;
+}
+
+async fn get_signed_tx(chain_id: u64,
+address: H160, receiver: H160, chunk:usize, gas_price: U256, provider, nonce: U256)-> Result<Bytes, WalletError> {
```

Comment:
`provider:&Arc<SignerMiddleware<M, LocalWallet>>` ?

Reply:
ohh thanks lemem try that.  simpler than copy pasta the giant thing around.

Reply:
```
cannot find type `M` in this scope
not found in this scoperustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message?2#file%3A%2F%2F%2FUsers%2Fdanlee%2Fcode%2Fstress4844%2Fsrc%2Fbundle_builder.rs)
bundle_builder.rs(38, 23): you might be missing a type parameter: `<M>`
```
hmm not used to generics in rust